xds/internal/client/transport_helper.go

```diff
@@ -246,10 +246,10 @@ func (t *TransportHelper) send(ctx context.Context) {
 			t.sendCh.Load()
 
 			var (
-				target         []string
-				rType          ResourceType
-				version, nonce string
-				send           bool
+				target                 []string
+				rType                  ResourceType
+				version, nonce, errMsg string
```

Comment:
Nit: We might as well get rid of `errMsg` and use `update.errMsg` directly.

Reply:
There's a `switch: case`. Only `ackAction` has the `errMsg` field.